src/convertToObject.js

```diff
@@ -3,16 +3,29 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example
+ * in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  //  write your code here
+  const rezult = {};
+  const stringToArr = sourceString.split(';');
+  const arrSplit = stringToArr.map(item => item.split(':'));
+  const arrTrim = arrSplit.map(item => item.map(elem => elem.trim()));
+  const joinArr = arrTrim.filter(item => !item.every(items => items === ''));
```

Comment:
you can use them one by one to the same variable, in chains, no need to create a new constant each time

like this arr.map().filter()